src/mono/wasm/debugger/BrowserDebugProxy/MemberReferenceResolver.cs

```diff
@@ -119,13 +119,35 @@ public async Task<JObject> TryToRunOnLoadedClasses(string varName, CancellationT
                     }
                 }
                 var store = await proxy.LoadStore(sessionId, token);
-                foreach (var asm in store.assemblies)
+                var methodInfo = ctx.CallStack.FirstOrDefault(s => s.Id == scopeId)?.Method?.Info;
+                var classNameToFindWithNamespace =
+                    string.IsNullOrEmpty(methodInfo?.TypeInfo?.Namespace) ?
+                    classNameToFind :
+                    methodInfo.TypeInfo.Namespace + "." + classNameToFind;
+
+                var searchResult = await TryFindNameInAssembly(store.assemblies, classNameToFindWithNamespace);
+                if (searchResult == null)
+                    searchResult = await TryFindNameInAssembly(store.assemblies, classNameToFind);
+                if (searchResult != null)
+                    typeId = (int)searchResult;
+
+                async Task<int?> TryGetTypeIdFromName(string typeName, AssemblyInfo assembly)
                 {
-                    var type = asm.GetTypeByName(classNameToFind);
-                    if (type != null)
+                    var type = assembly.GetTypeByName(typeName);
+                    if (type == null)
+                        return null;
+                    return await sdbHelper.GetTypeIdFromToken(sessionId, assembly.DebugId, type.Token, token);
+                }
+
+                async Task<int?> TryFindNameInAssembly(List<AssemblyInfo> assemblies, string name)
+                {
+                    foreach (var asm in assemblies)
                     {
-                        typeId = await sdbHelper.GetTypeIdFromToken(sessionId, asm.DebugId, type.Token, token);
+                        var typeId = await TryGetTypeIdFromName(name, asm);
+                        if (typeId != null)
+                            return typeId;
                     }
+                    return null;
```

Comment:
It makes sense to have these two combined into one method, but I know that you have another PR open about this, so this can be handled there.